vortex-layout/src/layouts/row_id.rs

```diff
@@ -75,18 +69,12 @@ impl RowIdLayoutReader {
 
         (
             ROW_ID.clone(),
-            StructArray::from_fields(&[
-                (
-                    "file_row_number",
-                    SequenceArray::typed_new(row_range.start, 1, arr_len)
-                        .vortex_expect("cannot be out of bounds")
-                        .to_array(),
-                ),
-                (
-                    "file_index",
-                    ConstantArray::new(self.file_index, arr_len).to_array(),
-                ),
-            ])
+            StructArray::from_fields(&[(
+                "file_row_number",
+                SequenceArray::typed_new(row_range.start, 1, arr_len)
+                    .vortex_expect("cannot be out of bounds")
+                    .to_array(),
+            )])
```

Comment:
IMO, the struct is less obvious than having the row index evaluate directly to the number.